### PR TITLE
[CI] Add comprehensive testing: migration, e2e, and bench

### PR DESCRIPTION
**Pull request by KuilongCui**

### CODE REVIEW

File: .github/workflows/migration.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: MIGRAION
```
Comment (zhypku) [inline .github/workflows/migration.yml:1]: typo

File: llumnix/backends/vllm/worker.py [hunk 2]
```diff
@@ -93,6 +95,7 @@ def init_migration(self, instance_id: str, migration_config: MigrationConfig, sr
         self.instance_id = instance_id
         self.global_world_size = 0
         self.global_rank = -1
+        # self.migration_config = migration_config
```
Comment (zhypku) [inline llumnix/backends/vllm/worker.py:98]: remove

File: .github/workflows/migration.yml (hunk 1 above)
Comment (KuilongCui) [reply .github/workflows/migration.yml]: get

File: llumnix/backends/vllm/worker.py (hunk 2 above)
Comment (KuilongCui) [reply llumnix/backends/vllm/worker.py]: get

File: conftest.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# conftest.py
```
[the remark this answers is not in the document]
Comment (KuilongCui) [reply conftest.py]: get

File: llumnix/backends/vllm/worker.py [hunk 4]
```diff
@@ -104,10 +107,19 @@ def init_migration(self, instance_id: str, migration_config: MigrationConfig, sr
...
             self.migration_backend.migrate_cache(src_worker_handle, src_blocks, dst_blocks)
         except ray.exceptions.RayActorError:
             logger.info("[migrate_cache] self.rank: {}, src_worker_handle {} is dead".format(self.rank, src_worker_handle))
+        end_time = time.time()
+
+        total_kv_cache_size = len(src_blocks) * CacheEngine.get_cache_block_size(
+            self.cache_config, self.model_config, self.parallel_config)
+        speed = total_kv_cache_size/1024/1024/1024/(end_time - start_time)
+        logger.info("[migration_cache] blocks_num: {}, total_kv_cache_size: {}, time: {}s, speed: {}GB/s."
+                    .format(len(src_blocks), convert_bytes(total_kv_cache_size), end_time-start_time, speed))
```
[the remark this answers is not in the document]
Comment (KuilongCui) [reply llumnix/backends/vllm/worker.py:122]: get

File: tests/unit_test/entrypoints/vllm/test_api_server.py [hunk 5]
```diff
@@ -59,6 +60,7 @@ def api_server():
     # Waiting for api server subprocess to terminate.
     time.sleep(1.0)
 
+# @pytest.mark.skip(reason="This test is flaky")
```
Comment (s5u13b) [inline tests/unit_test/entrypoints/vllm/test_api_server.py:63]: what does it mean? is it necessary now?

File: .github/workflows/bench.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
+name: BENCH
```
[the remark this answers is not in the document]
Comment (KuilongCui) [reply .github/workflows/bench.yml:1]: time-out-limit already exist for each test.
all the name in workflows will chang to the style "bench_test"

File: conftest.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2024, Alibaba Group;
```
[the remark this answers is not in the document]
Comment (KuilongCui) [reply conftest.py:1]: The conftest.py file is a special configuration file used to define shared fixtures, hooks, and other configurations in pytest. I will move it to the tests directory, and setup_ray_env will be put in it.

File: llumnix/entrypoints/llumnix_utils.py [hunk 8]
```diff
@@ -38,10 +39,9 @@
 MAX_TASK_RETRIES = 300
 RETRIES_INTERVALS = 0.1
 
-
 def get_ip_address():
-    result = subprocess.run(['hostname', '-i'], stdout=subprocess.PIPE, check=True)
-    ip_address = result.stdout.decode('utf-8').strip()
+    hostname = socket.gethostname()
```
[the remark this answers is not in the document]
Comment (KuilongCui) [reply llumnix/entrypoints/llumnix_utils.py:43]: hostname -i may sometimes return multiple addresses, which does not meet expectations.

File: tests/unit_test/entrypoints/vllm/test_api_server.py (hunk 5 above)
Comment (KuilongCui) [reply tests/unit_test/entrypoints/vllm/test_api_server.py:63]: I will delete it

File: conftest.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (s5u13b) [reply conftest.py]: the filename conftest.py is also a bit odd, why not just name it as utils.py?

[the remark this answers is not in the document]
Comment (KuilongCui) [reply conftest.py]: The name of conftest.py is a well-established convention in pytest.

**Review by zhypku — APPROVED** [review]

**Review by s5u13b — APPROVED** [review]

**Pull request merged**